src/drivers/i2c/i2c.h

```diff
@@ -1,27 +1,62 @@
 #ifndef DRIVERS_I2C_H_
 #define DRIVERS_I2C_H_
 
-struct i2c_device {
-	struct i2c_device *next;
+#include <stdint.h>
+
+#include <util/dlist.h>
+#include <framework/mod/options.h>
+#include <config/embox/driver/i2c.h>
+
+#define I2C_BUS_MAX   \
+	OPTION_MODULE_GET(embox__driver__i2c, NUMBER, i2c_bus_max)
+
+#define MAX_I2C_BUS_NAME    8
+
+struct i2c_algorithm;
+
+struct i2c_msg {
+	uint16_t addr;	/* slave address			*/
+	uint16_t flags;
+#define I2C_M_RD		0x0001	/* read data, from slave to master */
+					/* I2C_M_RD is guaranteed to be 0x0001! */
+#define I2C_M_TEN		0x0010	/* this is a ten bit chip address */
+#define I2C_M_RECV_LEN		0x0400	/* length will be first received byte */
+#define I2C_M_NO_RD_ACK		0x0800	/* if I2C_FUNC_PROTOCOL_MANGLING */
+#define I2C_M_IGNORE_NAK	0x1000	/* if I2C_FUNC_PROTOCOL_MANGLING */
+#define I2C_M_REV_DIR_ADDR	0x2000	/* if I2C_FUNC_PROTOCOL_MANGLING */
+#define I2C_M_NOSTART		0x4000	/* if I2C_FUNC_NOSTART */
+#define I2C_M_STOP		0x8000	/* if I2C_FUNC_PROTOCOL_MANGLING */
+	uint16_t len;		/* msg length				*/
+	uint8_t *buf;		/* pointer to msg data			*/
 };
 
 struct i2c_adapter {
-	void *priv;
+	void *i2c_algo_data;
+	const struct i2c_algorithm *i2c_algo; /* the algorithm to access the bus */
+};
+
+struct i2c_algorithm {
```

Comment:
I think `i2c_ops`/`i2c_operations` is more appropriate name for such structure.

Reply:
@Deryugin Yes, but it uses in Linux kernel, and I resided that it can be usefull